### PR TITLE
Feat: Build Pipeline

### PR DESCRIPTION
**Pull request by markcsims**
https://asaphub.atlassian.net/browse/CRN-606

This is the build pipeline for PRs. It's the build, test and check up to but not including Squidex Sync.

- [x] Build
- [x] Rebuild
- [x] Unit Test
- [x] Integration Test
- [x] Test Build Output
- [x] Test Browser
- [x] Bundlewatch
- [x] Check Constraints
- [x] Check Format
- [x] Check Packages
- [x] Serverless Package

Moved to seperate PR:

- Prepare Squidex
- Verify Packages
- Notify on Error

### CODE REVIEW

**Comment by marot** [comment]
Should we remove the duplicate jobs in .gitlab-ci.yml in context of this PR?

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -30,7 +30,11 @@ jobs:
           IMAGE_NAME=$ECR_REGISTRY/$ECR_REPOSITORY
           docker build -t $IMAGE_NAME:$IMAGE_TAG . -f ci/Dockerfile --target base
           docker tag $IMAGE_NAME:$IMAGE_TAG $IMAGE_NAME:latest
-          docker push --all-tags $IMAGE_NAME
+          echo "IMAGE_NAME=$IMAGE_NAME" >> $GITHUB_ENV
```
Comment (marot) [inline .github/workflows/build-images.yml:33]: Can we move this to line 28?

File: .github/workflows/build-images.yml [hunk 2]
```diff
@@ -30,7 +30,11 @@ jobs:
...
+      - name: push squidex
+        if: ${{ github.event_name == 'workflow_dispatch' }}
```
Comment (marot) [inline .github/workflows/build-images.yml:35]: Why does this only happen on workflow dispatch? What happens to the images on master?

File: .github/workflows/build-images.yml [hunk 3]
```diff
@@ -40,4 +44,7 @@ jobs:
           IMAGE_NAME=$ECR_REGISTRY/$ECR_REPOSITORY
           docker build -t $IMAGE_NAME:$IMAGE_TAG . -f ci/Dockerfile --target integration
           docker tag $IMAGE_NAME:$IMAGE_TAG $IMAGE_NAME:latest
-          docker push --all-tags $IMAGE_NAME
+          echo "IMAGE_NAME=$IMAGE_NAME" >> $GITHUB_ENV
```
Comment (marot) [inline .github/workflows/build-images.yml:47]: same two comments here

File: .github/workflows/setup.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        type: string
+      image-tag:
+        description: 'Image Tag'
+        required: false
+        default: 3832bee3b58efae7fb29e5c8517140300c4c9daf
+        type: string
+      node-version:
+        description: 'Node Version'
+        required: false
+        default: '14.17'
```
Comment (marot) [inline .github/workflows/setup.yml:24]: shouldnt this be 14.19? Does it make sense to keep the default values here?

File: .github/workflows/setup.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        description: 'Name of integration image'
+        value: ${{ jobs.image-names.outputs.integration-image-name }}
+      squidex-image-name:
+        description: 'Name of Squidex image'
+        value: ${{ jobs.image-names.outputs.squidex-image-name }}
+      node-image-name:
+        description: 'Name of node image'
+        value: ${{ jobs.image-names.outputs.node-image-name }}
+jobs:
+  image-names:
```
Comment (marot) [inline .github/workflows/setup.yml:40]: I remember this loading some data by logging into aws. Since this is gone now, can't we simplify this as a shell script?

File: .github/workflows/build-pipeline.yml [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        uses: actions/checkout@v2
+      - name: Cache Jest cache
+        uses: actions/cache@v2
+        id: jest-cache
+        with:
+          path: |
+            .jest-cache/
+          key: jest-cache
+      - name: Test
+        run: yarn test --coverage --maxWorkers=2
```
Comment (marot) [inline .github/workflows/build-pipeline.yml:31]: The gitlab version also has `yarn codecov`. Isn't that necessary anymore?

File: .github/workflows/build-pipeline.yml [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+        run: |
+          REACT_APP_ALGOLIA_APP_ID=$(aws ssm get-parameter --name "algolia-app-id-dev" --query Parameter.Value --output text)
+          echo "::set-output name=id::$REACT_APP_ALGOLIA_APP_ID"
+      - name: Build
+        env:
+          NODE_ENV: 'production'
+          ASAP_HOSTNAME: 'hub.asap.science'
+          ASAP_API_URL: 'https://api-${ needs.setup.outputs.pr-number }.${ASAP_HOSTNAME}'
+          ASAP_APP_URL: 'https://${ needs.setup.outputs.pr-number }.${ASAP_HOSTNAME}'
+          ALGOLIA_INDEX: 'asap-hub_research_outputs_CI-${ needs.setup.outputs.pr-number }'
```
Comment (marot) [inline .github/workflows/build-pipeline.yml:84]: should be asap-hub_CI-${...} now

File: .github/workflows/build-pipeline.yml [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+      credentials:
+        username: AWS
+        password: ${{ secrets.AWS_ECR_PASSWORD }}
+    env:
+      SLS_STAGE: ${{ needs.setup.outputs.pr-number }}
+      NODE_ENV: 'production'
+      ASAP_HOSTNAME: 'hub.asap.science'
+      ASAP_API_URL: 'https://api-${{ needs.setup.outputs.pr-number }}.hub.asap.science'
+      ASAP_APP_URL: 'https://${{ needs.setup.outputs.pr-number }}.hub.asap.science'
+      ALGOLIA_INDEX: 'asap-hub_research_outputs_CI-${{ needs.setup.outputs.pr-number }}'
```
Comment (marot) [inline .github/workflows/build-pipeline.yml:160]: same here

File: apps/frontend/src/network/users/__tests__/UserProfile.test.tsx [hunk 9]
```diff
@@ -115,6 +115,7 @@ const renderUserProfile = async (
   return result;
 };
 
+jest.retryTimes(3);
```
Comment (marot) [inline apps/frontend/src/network/users/__tests__/UserProfile.test.tsx:118]: why is this necessary?

**Review by kmaid — APPROVED** [review]

**Review by peterstarling — APPROVED** [review]
